lib/std/crypto/25519/ed25519.zig

```diff
@@ -214,17 +219,26 @@ pub const Ed25519 = struct {
         }
 
         /// Create a Verifier for incremental verification of a signature.
-        pub fn verifier(self: Signature, public_key: PublicKey) (NonCanonicalError || EncodingError || IdentityElementError)!Verifier {
-            return Verifier.init(self, public_key);
+        pub fn verifier(sig: Signature, public_key: PublicKey) Verifier.InitError!Verifier {
+            return Verifier.init(sig, public_key);
         }
 
+        pub const VerifyError = Verifier.InitError || Verifier.VerifyError;
+
         /// Verify the signature against a message and public key.
         /// Return IdentityElement or NonCanonical if the public key or signature are not in the expected range,
         /// or SignatureVerificationError if the signature is invalid for the given message and key.
-        pub fn verify(self: Signature, msg: []const u8, public_key: PublicKey) (IdentityElementError || NonCanonicalError || SignatureVerificationError || EncodingError || WeakPublicKeyError)!void {
-            var st = try Verifier.init(self, public_key);
-            st.update(msg);
-            return st.verify();
+        pub fn verify(sig: Signature, msg: []const u8, public_key: PublicKey) VerifyError!void {
+            try sig.concatVerify(&.{msg}, public_key);
+        }
+
+        /// Verify the signature against a concatenated message and public key.
+        /// Return IdentityElement or NonCanonical if the public key or signature are not in the expected range,
+        /// or SignatureVerificationError if the signature is invalid for the given message and key.
+        pub fn concatVerify(sig: Signature, msg: []const []const u8, public_key: PublicKey) VerifyError!void {
```

Comment:
As this is very protocol specific, maybe the `tis` namespace would be a better place for that function?